src/operator/mxnet_op.h

```diff
@@ -88,6 +88,51 @@ struct clip_grad {
   }
 };
 
+#define REVERSE_MAX_DIM 10
+
+struct reverse {
+  MSHADOW_XINLINE static int ReverseIndex(index_t idx,
+                                          const index_t nreversedim,
+                                          const index_t * stride_,
+                                          const index_t * trailing_) {
+    index_t outputIndex = idx;
+    for (index_t i = 0; i < nreversedim; ++i) {
+      const index_t low = outputIndex % trailing_[i];
+      index_t high = outputIndex / trailing_[i];
+      const index_t x = high%stride_[i];
+      high /= stride_[i];
+      outputIndex = (high*stride_[i] + stride_[i] - 1 - x)*trailing_[i] + low;
+    }
+    return outputIndex;
+  }
+#ifdef __CUDACC__
+  template<typename DType>
+  __device__  static void Map(int index, const index_t nreversedim,
+                              const DType *src, DType *dst,
+                              const index_t * stride_,
+                              const index_t * trailing_) {
+    __shared__ index_t stride_share[REVERSE_MAX_DIM];
+    __shared__ index_t trailing_share[REVERSE_MAX_DIM];
+    if (threadIdx.x < REVERSE_MAX_DIM) {
```

Comment:
NV guys said you need copying because nreversedim is not a constant. If you make it a template argument and switch before launch you'll be fine with Shape<ndim>